sys/linux/dev_bifrost.txt

```diff
@@ -0,0 +1,370 @@
+	version_buffer	ptr64[out, array[int8]]
+	size		len[version_buffer, int32]
+	padding		const[0, int32]
+}
+
+kbase_ioctl_mem_jit_init_10_2 {
+	va_pages	int64
+}
+
+kbase_ioctl_mem_jit_init_11_5 {
+	va_pages	int64
+	max_allocations	int8
+	trim_level	int8
+	group_id	int8
+	padding		array[const[0, int8], 5]
+}
+
+kbase_ioctl_mem_jit_init {
+	va_pages	int64
+	max_allocations	int8
+	trim_level	int8
+	group_id	int8
+	padding		array[const[0, int8], 5]
+	phys_pages	int64
+}
+
+kbase_ioctl_mem_sync {
+	handle		gpu_addr
+	user_addr	user_addr
+	size		int64
+	type		flags[base_syncset_op_flags, int8]
+	padding		array[const[0, int8], 7]
+}
+
+kbase_ioctl_mem_find_cpu_offset [
+	in	kbase_ioctl_mem_find_cpu_offset_in	(in)
+	out	kbase_ioctl_mem_find_cpu_offset_out	(out)
+]
+
+kbase_ioctl_mem_find_cpu_offset_in {
+	gpu_addr	gpu_addr
+	cpu_addr	user_addr
+	size		int64
+}
+
+kbase_ioctl_mem_find_cpu_offset_out {
+	offset	int64
+}
+
+kbase_ioctl_get_context_id {
+	id	int32
+}
+
+kbase_ioctl_tlstream_acquire {
+	flags	int32
+}
+
+kbase_ioctl_mem_commit {
+	gpu_addr	gpu_addr
+	pages		int64
+}
+
+kbase_ioctl_mem_alias [
+	in	kbase_ioctl_mem_alias_in	(in)
+	out	kbase_ioctl_mem_alias_out	(out)
+]
+
+kbase_ioctl_mem_alias_in {
+	flags		flags[base_mem_alloc_flags, int64]
+	stride		int64
+	nents		len[aliasing_info, int64]
+	aliasing_info	ptr64[in, array[base_mem_aliasing_info]]
+}
+
+kbase_ioctl_mem_alias_out {
+	flags		int64
+	gpu_va		gpu_addr
+	va_pages	int64
+}
+
+base_mem_aliasing_info {
+	handle	gpu_addr
+	offset	int64
+	length	int64
+}
+
+kbase_ioctl_mem_import [
+	in	kbase_ioctl_mem_import_in	(in)
+	out	kbase_ioctl_mem_import_out	(out)
+]
+
+kbase_ioctl_mem_import_in {
+	flags	flags[base_mem_alloc_flags, int64]
+	phandle	int64
+	type	flags[base_mem_import_type, int32]
+	padding	int32
+}
+
+kbase_ioctl_mem_import_out {
+	flags		int64
+	gpu_va		gpu_addr
+	va_pages	int64
+}
+
+kbase_ioctl_mem_flags_change {
+	gpu_va	gpu_addr
+	flags	flags[base_mem_alloc_flags, int64]
+	mask	int64
+}
+
+kbase_ioctl_stream_create {
+	name	array[int8, 32]
+}
+
+kbase_ioctl_fence_validate {
+	fd	fd_fence
+}
+
+kbase_ioctl_mem_profile_add {
+	buffer	ptr64[in, array[int8]]
+	len	len[buffer, int32]
+	padding	int32
+}
+
+kbase_ioctl_sticky_resource_map {
+	count	len[address, int64]
+	address	ptr64[in, array[int64]]
+}
+
+kbase_ioctl_sticky_resource_unmap {
+	count	len[address, int64]
+	address	ptr64[in, array[int64]]
+}
+
+kbase_ioctl_mem_find_gpu_start_and_offset [
+	in	kbase_ioctl_mem_find_gpu_start_and_offset_in	(in)
+	out	kbase_ioctl_mem_find_gpu_start_and_offset_out	(out)
+]
+
+kbase_ioctl_mem_find_gpu_start_and_offset_in {
+	gpu_addr	gpu_addr
+	size		int64
+}
+
+kbase_ioctl_mem_find_gpu_start_and_offset_out {
+	start	gpu_addr
+	offset	int64
+}
+
+kbase_ioctl_mem_exec_init {
+	va_pages	int64
+}
+
+kbase_ioctl_get_cpu_gpu_timeinfo [
+	in	kbase_ioctl_get_cpu_gpu_timeinfo_in	(in)
+	out	kbase_ioctl_get_cpu_gpu_timeinfo_out	(out)
+]
+
+kbase_ioctl_get_cpu_gpu_timeinfo_in {
+	request_flags	int32
```

Comment:
There seem to be some specific values for this such as BASE_TIMEINFO_MONOTONIC_FLAG, BASE_TIMEINFO_TIMESTAMP_FLAG, BASE_TIMEINFO_CYCLE_COUNTER_FLAG, BASE_TIMEINFO_KERNEL_SOURCE_FLAG, BASE_TIMEINFO_USER_SOURCE_FLAG.

Reply:
Fixed.